hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/infra/EthereumCallDataHydration.java

```diff
@@ -74,9 +74,19 @@ public HydratedEthTxData tryToHydrate(
             if (callDataFile.deleted()) {
                 return failureFrom(FILE_DELETED);
             }
+
+            final var initcodeBytes = callDataFile.contents().toByteArray();
```

Comment:
If you step back and look at L78-L92 what you're really trying to do is convert the file contents, which is in pbj's `Byte` class, to a `byte[]`.  And the only problem is that the first two bytes of the pbj-encoded byte array may in fact be the characters `0` and `x` - and, say, who the heck allowed that?

Oh well.  Anyway, all you _really_ want to do is take a slice of the original pbj `Bytes` iff the first two characters are the hex prefix, or else the entire array if it is not.  So why not just use the `Bytes.getBytes(long,long)` overload to do the slice (very cheap) and then `Bytes.toByteArray()` to get the contents as a `byte[]`?

```java

            final var hexPrefix = new byte[]{(byte)'0', (byte)'x'};
            final var contents = callDataFile.contents();
            final var offset = contents.matchesPrefix(hexPrefix) ? hexPrefix.length : 0L;
            final var len = contents.length() - offset;
            final var callData = contents.getBytes(offset,len).toByteArray();
```

You'll need to check that actually works though - it syntax checks but that doesn't mean it's correct.

Reply:
I just had to wrap the `callData` in `Hex.decode`, other then that it works perfectly 💯